secrets.sh

```diff
@@ -1,363 +1,496 @@
-      echo "Error: Chart package required."
-      exit 1
+clean() {
+    if is_help "$1"
+    then
+	clean_usage
+	return
     fi
-    enc "$2"
+    local basedir="$1"
+    find "$basedir" -type f -name "*${DEC_SUFFIX}" -print0 | xargs -r0 rm -v
+}
+
+helm_wrapper() {
+    local cmd="$1" subcmd='' cmd_version=''
     shift
-    ;;
-  "dec"):
-    if [[ $# -lt 2 ]]; then
-      dec_usage
-      echo "Error: Chart package required."
-      exit 1
-    fi
-    dec "$2"
-    ;;
-  "clean"):
-    if [[ $# -lt 2 ]]; then
-      clean_usage
-      echo "Error: Chart package required."
-      exit 1
+    if [[ $cmd == diff ]]
+    then
+	subcmd="$1"
+	shift
+	cmd_version=$(${HELM_BIN} diff version)
     fi
-    clean "$2"
-    ;;
-  "dec-deps"):
-    if [[ $# -lt 2 ]]; then
-      dec_deps_usage
-      echo "Error: Chart package required."
-      exit 1
-    fi
-    dec_deps "$2"
-    ;;
-  "view"):
-    if [[ $# -lt 2 ]]; then
-      view_usage
-      echo "Error: Chart package required."
-      exit 1
+
+    # cache options for the helm command in a file so we don't need to parse the help each time
+    local helm_version=$(${HELM_BIN} version --client --short)
+    local cur_options_version="${helm_version}${cmd_version:+ $cmd: }${cmd_version}"
+    local optfile="$HELM_PLUGIN_DIR/helm.${cmd}${subcmd:+.}${subcmd}.options" options_version='' options='' longoptions=''
+    [[ -f $optfile ]] && . "$optfile"
+
+    if [[ $cur_options_version != $options_version ]]
+    then
+	local re='(-([a-zA-Z0-9]), )?--([-_a-zA-Z0-9]+)( ([a-zA-Z0-9]+))?' line
+	options='' longoptions=''
+
+	# parse the helm options and option args from the help output
+	while read line
+	do
+	    if [[ $line =~ $re ]]
+	    then
+		local opt="${BASH_REMATCH[2]}" lopt="${BASH_REMATCH[3]}" optarg="${BASH_REMATCH[5]:+:}"
+		[[ $opt ]] && options+="${opt}${optarg}"
+		[[ $lopt ]] && longoptions+="${longoptions:+,}${lopt}${optarg}"
+	    fi
+	done <<<"$(${HELM_BIN} "$cmd" $subcmd --help | sed -e '1,/^Flags:/d')"
+
+	cat >"$optfile" <<EOF
+options_version='$cur_options_version'
+options='$options'
+longoptions='$longoptions'
+EOF
     fi
-    view "$2"
-    ;;
-  "edit"):
-    if [[ $# -lt 2 ]]; then
-      edit_usage
-      echo "Error: Chart package required."
-      exit 1
+    
+    # parse command line
+    local parsed # separate line, otherwise the return value of getopt is ignored
+    # if parsing fails, getopt returns non-0, and the shell exits due to "set -e"
+    parsed=$(getopt --options="$options" --longoptions="$longoptions" --name="${HELM_BIN} $cmd${subcmd:+ }$subcmd" -- "$@")
+
+    # collect cmd options with optional option arguments
+    local -a cmdopts=() decfiles=()
+    local yml ymldec decrypted
+    eval set -- "$parsed"
+    while [[ $# -gt 0 ]]
+    do
+	case "$1" in
+	    --)
+		# skip --, and what remains are the cmd args
+		shift 
+		break
+		;;
+            -f|--values)
+		cmdopts+=("$1")
+		yml="$2"
+		if [[ $yml =~ ^(.*/)?secrets(\.[^.]+)\.yaml$ ]]
```

Comment:
can we make this
`if [[ $yml =~ ^(.*/)?secrets(\.[^.]+)*\.yaml$ ]]`
to match `secrets.yaml` and `secrets.bla.yaml`
 
It seams to be a common practise to have Config / GitOps repo like: 
`<env>/<chartname>/{values,secrets}.yaml`
which could support this opinionated structure too.

Reply:
Thanks for spotting that!